### PR TITLE
Allow overriding MockProviderConfig for inline style, too (currently only ScalaTest, to be transferred for other frameworks).

### PR DESCRIPTION
**Pull request by danile42**
Hi,

first of all, great that you implemented the "inline style" for the other frameworks, too! :)

Unfortunately, the use of internal forger classes (ForgerImpl) made it no longer possible to override mockProviderConfig in tests with the "inline style" (as documented here: https://github.com/jbwheatley/pact4s#choosing-a-port).

With my changes, this is possible (again) for ScalaTest. If you approve this solution, I would try to do the same for the other frameworks, which should be straightforward.

I tried other solutions, but chose this one, mainly because it provides the same API for users of pact4s, regardless of which style (inline or not) is used.

I ran scalafmtAll as required, but since I'm on Windows, this produced a lot of changes in line separators only, which I reverted.
Also, commitCheck was not available for me.

Best regards,
Daniel

### CODE REVIEW

**Review by jbwheatley — CHANGES_REQUESTED** [review]
Hi Danile, thanks for the contribution! Sorry for inadvertently creating a regression when porting this feature. 

I think the cleanest solution that should work for all the implementations is to just add the config to the `InlineRequestResponsePactResources`

```scala
trait InlineRequestResponsePactResources extends RequestResponsePactOps { self =>
  def pactTestExecutionContext: PactTestExecutionContext = new PactTestExecutionContext()
  val mockProviderConfig: MockProviderConfig             = MockProviderConfig.createDefault()

  private[pact4s] type Forger <: InlineRequestResponsePactForger

  def withPact(aPact: RequestResponsePact): Forger

  private[pact4s] type Effect[_]

  /** This effect runs after each consumer pact test are run, but before they get written to a file.
    */
  def beforeWritePacts(): Effect[Unit]

  private[pact4s] trait InlineRequestResponsePactForger extends RequestResponsePactForgerResources {
    override private[pact4s] type Effect[A] = Id[A]
    override def beforeWritePacts(): Effect[Unit]       = ()
    override val mockProviderConfig: MockProviderConfig = self.mockProviderConfig
  }
}
```

Then because all the different instances of `ForgerImpl` extend that shared `InlineRequestResponsePactForger`, they all get the shared config.

**Comment by jbwheatley** [comment]
Also good catch about commitCheck.. I seem to have removed it 😓 `quickCommitCheck` remains, though its not that quick... I'll either update the contributor guide or add the cross compilng `commitCheck` back.

**Comment by jbwheatley** [comment]
As for the scalafmt thing - if you add `lineEndings=preserve` to the config does that fix it? If it does please include it in the PR for our other windows users!

File: shared/src/main/scala/pact4s/BasePactResources.scala [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package pact4s
+
+import au.com.dius.pact.consumer.PactTestExecutionContext
+import au.com.dius.pact.consumer.model.MockProviderConfig
+
+trait BasePactResources {
```
Comment (jbwheatley) [inline shared/src/main/scala/pact4s/BasePactResources.scala:22]: I think we don't need this extra abstraction, there's only two classes that need the config and the execution context so its less of a change to just add the mockConfig to `InlineRequestResponsePactResources`.

Comment (danile42) [reply shared/src/main/scala/pact4s/BasePactResources.scala]: Okay, I removed it.

**Comment by danile42** [comment]
> > [error] C:\source\pact4s\models\src\main\scala\pact4s\provider\ProviderInfoBuilder.scala:269:3: Could not find any member to link for "java.net.URL".
> > [error] /** Create a ProviderInfoBuilder by providing a [[java.net.URL]] rather than specifying the URL components separately
> > [error] ^
> > [error] one error found
> 
> I'm curious, what version of java are you running locally?

Microsoft JDK 17.0.6.

**Review by jbwheatley — APPROVED** [review]
thanks!

**Comment by jbwheatley** [comment]
> > > [error] C:\source\pact4s\models\src\main\scala\pact4s\provider\ProviderInfoBuilder.scala:269:3: Could not find any member to link for "java.net.URL".
> > > [error] /** Create a ProviderInfoBuilder by providing a [[java.net.URL]] rather than specifying the URL components separately
> > > [error] ^
> > > [error] one error found
> > 
> > 
> > I'm curious, what version of java are you running locally?
> 
> Microsoft JDK 17.0.6. Edit: I just changed to Oracle JDK 23.0.1, and the error is still reported.

Yeah can't recreate this on my machine, even using the same jdk. For reference i'm on arm64 running sbt from the command line.

**Comment by jbwheatley** [comment]
its the `sbt-api-mappings` plugin that does the linking

**Pull request merged**